v-next/hardhat/src/internal/builtin-plugins/network-manager/type-extensions.ts

```diff
@@ -0,0 +1,164 @@
+import type { NetworkConnection, NetworkManager } from "./types.js";
```

Comment:
Just a tip: this file can also be split into multiple files.

Maybe something like: 

```
type-extensions/
  index.ts -- imports the rest
  config.ts
  hooks.ts
```

Reply:
I like the suggestion, I'll take it 👍